vnext/Chakra/ChakraJsiRuntime_core.cpp

```diff
@@ -334,32 +254,70 @@ JsErrorCode ChakraJsiRuntimeWithDebugger::enableDebugging(JsRuntimeHandle runtim
   return result;
 }
 
-/* static */ void ChakraJsiRuntimeWithDebugger::ProcessDebuggerCommandQueueCallback(void* callbackState) {
-  ChakraJsiRuntimeWithDebugger* runtime= reinterpret_cast<ChakraJsiRuntimeWithDebugger*>(callbackState);
+/* static */ void ChakraJsiRuntime::ProcessDebuggerCommandQueueCallback(void* callbackState) {
+  ChakraJsiRuntime* runtime = reinterpret_cast<ChakraJsiRuntime*>(callbackState);
 
   if (runtime) {
     runtime->ProcessDebuggerCommandQueue();
   }
 }
 
-void ChakraJsiRuntimeWithDebugger::ProcessDebuggerCommandQueue() {
+void ChakraJsiRuntime::ProcessDebuggerCommandQueue() {
   if (runtimeArgs().jsQueue) {
     runtimeArgs().jsQueue->runOnQueue([this]() {
-      if (debugProtocolHandler_) {
-        debugProtocolHandler_->ProcessCommandQueue();
+      if (m_debugProtocolHandler) {
+        m_debugProtocolHandler->ProcessCommandQueue();
       }
     });
   }
 }
 
-#endif
+void ChakraJsiRuntime::startDebuggingIfNeeded() {
 
-std::unique_ptr<jsi::Runtime> makeChakraJsiRuntime(ChakraJsiRuntimeArgs&& args) noexcept {
-#ifdef CHAKRART_HAS_DEBUGGER
-  return std::make_unique<ChakraJsiRuntimeWithDebugger>(std::move(args));
-#else
-  return std::make_unique<ChakraJsiRuntime>(std::move(args));
-#endif
+  if (runtimeArgs().enableDebugging)
+  {
+    std::string runtimeName = runtimeArgs().debuggerRuntimeName;
+    int port = runtimeArgs().debuggerPort;
+
+    if (runtimeName.empty()) {
+      runtimeName = DebuggerDefaultRuntimeName;
+    }
+
+    if (port == 0) {
+      port = DebuggerDefaultPort;
+    }
```

Comment:
auto args = runtimeArgs();
auto port = args.debuggerPort == 0 ? DebuggerDefaultPort: args.debuggerPort;
auto runtimeName = args.debuggerRuntimeName.empty() ? DebuggerDefaultPort : args.debuggerRuntimeName;

Reply:
Thanks for all these suggestions .. These code are copied from the older chakraexecutor stack .. I was planning a cleanup later ..